client/src/app/pages/applications/applications-table/components/manage-columns-toolbar.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  Button,
+  OverflowMenu,
+  OverflowMenuGroup,
+  OverflowMenuItem,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { ColumnState } from "@app/hooks/table-controls/column/useColumnState";
+import { ManageColumnsModal } from "./manage-columns-modal";
+
+// Define props to accept columns and setColumns directly
+interface ManageColumnsToolbarProps<TColumnKey extends string> {
+  columns: ColumnState<TColumnKey>[];
+  setColumns: (newColumns: ColumnState<TColumnKey>[]) => void;
+}
+
+export const ManageColumnsToolbar = <TColumnKey extends string>({
+  columns,
+  setColumns,
+}: ManageColumnsToolbarProps<TColumnKey>) => {
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      <ToolbarItem variant="overflow-menu">
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuGroup groupType="button" isPersistent>
+            <OverflowMenuItem isPersistent>
+              <Button variant="link" onClick={() => setIsOpen(true)}>
+                Manage columns
```

Comment:
how about an icon? there is a dedicated ColumnsIcon -> https://www.patternfly.org/design-foundations/icons/#all-icons
>Indicates the ability to manage columns for a table view
>

Reply:
> how about an icon? there is a dedicated ColumnsIcon -> https://www.patternfly.org/design-foundations/icons/#all-icons
> 
> > Indicates the ability to manage columns for a table view

i’m slightly indifferent. I think the manage column icon is fine, the only issue for me is our users wont know what it is bc they havent used it. 

updated: I'm in a UX Face 2 Face... general consensus is use the icon to save space